kernel.go

```diff
@@ -244,8 +263,18 @@ func handleExecuteRequest(ir *classic.Interp, receipt msgReceipt) error {
 		ExecCounter++
 	}
 
+	// Prepare the map that will hold the reply content
+	content := make(map[string]interface{})
 	content["execution_count"] = ExecCounter
 
+	// Tell the front-end that the kernel is working and when finished notify the
+	// front-end that the kernel is idle again
+	receipt.PublishKernelBusy()
```

Comment:
I might prefer to just put the Publish method inline here for busy, idle, etc. messages.  This could be a one line call and would avoid hiding what's actually happening in these methods.  Also, I would handle the error here.  Looks like the error from Publish isn't handled in these methods.

Reply:
Also, many of these publish messages (busy, idle, etc.) appear to be identical minus the actual status.  Could you just pass in the status in as an argument and not duplicate the methods.  You could just have `PublishKernelStatus()`.

Reply:
The reasoning behind hiding the message being sent is to encapsulate the details of the protocol inside messages.go. This should make it easier to maintain when Jupyter updates again in the future. 

I agree with the second comment, I can clean up the kernel status messages.

Reply:
I guess I could see that separation of concerns.  Good point.